File: 7_multiple_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Add
{
public:
    void sum(double a, double b)
    {
        cout << " The sum of " << a << " and " << b << " is " << a + b << endl;
    }
};

class Mul
{
public:
    void mul(double a, double b)
    {
        cout << " The Multiplication of " << a << " and " << b << " is " << a * b << endl;
    }
};

class Sub
{
public:
    void sub(double a, double b)
    {
        cout << " The Subtraction of " << a << " and " << b << " is " << a - b << endl;
    }
};

class Div
{
public:
    void div(double a, double b)
    {
        cout << " The Division of " << a << " and " << b << " is " << a / b << endl;
    }
};

class Calculate : public Add, public Div, public Sub, public Mul
{
public:
    int p = 12;
    int q = 5;
    void complete()
    {
        cout << "The Calculation is Completed" << endl;
    }
};

int main()
{
    Calculate problem;
    problem.sum(2, 2);
    problem.mul(2, 2);
    problem.div(2, 2);
    problem.sub(2, 2);
}

File: 7_multiple_inheritance.py
class Add:
    def sum(self, a, b):
        print(f" The sum of {a} and {b} is {a + b}")


class Mul:
    def mul(self, a, b):
        print(f" The Multiplication of {a} and {b} is {a * b}")


class Sub:
    def sub(self, a, b):
        print(f" The Subtraction of {a} and {b} is {a - b}")


class Div:
    def div(self, a, b):
        print(f" The Division of {a} and {b} is {a / b}")


class Calculate(Add, Div, Sub, Mul):
    p = 12
    q = 5

    def complete(self):
        print("The Calculation is Completed")


problem = Calculate()
problem.sum(2, 2)
problem.mul(2, 2)
problem.div(2, 2)
problem.sub(2, 2)
